app/com/lynxanalytics/biggraph/frontend_operations/VertexAttributeOperations.scala

```diff
@@ -7,17 +7,87 @@ import com.lynxanalytics.biggraph.graph_api.Scripting._
 import com.lynxanalytics.biggraph.graph_operations
 import com.lynxanalytics.biggraph.graph_util.Scripting._
 import com.lynxanalytics.biggraph.controllers._
-import com.lynxanalytics.biggraph.graph_api.Attribute
-import com.lynxanalytics.biggraph.graph_api.Scalar
+import com.lynxanalytics.biggraph.graph_api._
+import com.lynxanalytics.biggraph.controllers.OperationParams._
 
+abstract class UseTableAsAttributeOperation(context: Operation.Context, ops: ProjectOperations)
+  extends ProjectOutputOperation(context) {
+  import Operation.Implicits._
+  override lazy val project = projectInput("graph")
+  lazy val table = tableLikeInput("attributes").asProject
+  def projectAttributes: StateMapHolder[Attribute[_]]
+  def projectIdSet: VertexSet
+  def tableAttributes = table.vertexAttributes
+  val projectAttrNames = projectAttributes.iterator.map(_._1).toList.sorted
+
+  params ++= List(
+    Choice("id_attr", "ID attribute",
+      options = FEOption.unset +: FEOption.list(projectAttrNames)),
+    Choice("id_column", "ID column",
+      options = FEOption.unset +: table.vertexAttrList),
+    Param("prefix", "Name prefix for the imported attributes"),
+    Choice("unique_keys", "Assert unique ID attribute values", options = FEOption.boolsDefaultFalse),
```

Comment:
Weren't we going to remove this? Turned out not to be easily possible?

Reply:
> Weren't we going to remove this? Turned out not to be easily possible?

What we have support for is hidden but allowed options for dropdowns. We don't have a mechanism for hidden but allowed parameters and I don't think this is import enough to add it. I just changed the default to `false`.

Reply:
OK!

Reply:
OK, sounds good!